map/hla_manager.cpp

```diff
@@ -0,0 +1,343 @@
+#include "map/hla_manager.hpp"
+
+#include "drape_frontend/drape_engine.hpp"
+#include "drape_frontend/visual_params.hpp"
+
+#include "indexer/scales.hpp"
+
+#include "platform/http_client.hpp"
+#include "platform/platform.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/file_name_utils.hpp"
+
+namespace
+{
+std::string const kCampaignFile = "hla_campaigns.dat";
+std::string const kExpirationFile = "hla_expiration.dat";
+auto constexpr kWWanUpdateTimeout = std::chrono::hours(12);
+
+std::vector<uint8_t> SerializeTimestamp(std::chrono::steady_clock::time_point ts)
+{
+  long hours = std::chrono::duration_cast<std::chrono::hours>(ts.time_since_epoch()).count();
+  std::vector<uint8_t> result;
+  uint8_t * bytes = reinterpret_cast<uint8_t * >(&hours);
```

Comment:
Again, please, don't use this - we have good primitives for writing/reading primitives.